<!-- Launch file for peg-transfer training excercise -->

<launch>
	<group ns="saf">

        <arg name="mode" default="execution"/>
            <!-- execution, calibration, acc_blocks, acc_pegs-->

        <arg name="board_descriptor"
        default="peg_transfer_board.yaml"/>

        <arg name="offset_arm_1"
        default="offset_psm_1_uni.yaml"/>

        <arg name="on_dist_threshold" default="5.0"/>

        <arg name="speed_cartesian" default="40.0"/>
        <arg name="speed_jaw" default="60.0"/>


        <arg name="measurement_filename"
            default="/home/tamas/data/pegransfer_block_err/peg_err_1.csv"/>

        <node name="raisins" pkg="irob_subtask_logic"
            type="raisins" output="screen">

                <param name="speed_cartesian" type="double" value="$(arg speed_cartesian)" />
                <param name="speed_jaw" type="double" value="$(arg speed_jaw)" />
                <param name="on_dist_threshold" type="double" value="$(arg on_dist_threshold)" />
                <param name="mode" type="string" value="$(arg mode)" />
                <param name="measurement_filename" type="string" value="$(arg measurement_filename)" />
                <rosparam param="arm_names">["arm_1"]</rosparam>
                <param name="offset_filename_arm_1" type="string" value="$(find irob_subtask_logic)/config/$(arg offset_arm_1)" />

                <rosparam command="load"
                file="$(find irob_subtask_logic)/config/$(arg board_descriptor)"/>

                <rosparam command="load"
                file="$(find irob_subtask_logic)/config/$(arg offset_arm_1)"/>

        <remap from="target" to="vision/blocks_grasp" />
	</node>
  	</group>
</launch>